src/main/java/com/shigotosagashi/jobdomainservice/repository/ExampleEntityRepository.java

```diff
@@ -4,12 +4,9 @@
 import org.springframework.data.repository.CrudRepository;
 
 import java.util.List;
-import java.util.Optional;
 
 public interface ExampleEntityRepository extends CrudRepository<ExampleEntity, String> {
 
-  Optional<ExampleEntity> findByTestId(String testId);
```

Comment:
Qual a vantagem de utilizar Optional, ao invés do objeto diretamente?

Reply:
**Optional** é uma forma mais elegante de tratar objetos nulos. Ele não previne um **NullPointerException**, por exemplo, porém nós conseguimos tomar decisões e executar métodos diretamente a partir deste objeto, sem a necessidade de escrever **if (objeto != null) {} else {}**

Um exemplo de uso deste objeto está na classe **JobController**, no método **findById**. A partir do **Optional**, nós conseguimos chamar o método **fromEntity** e retornar a resposta ao usuário, caso contrário uma exceção será lançada.
Se não tivesse **Optional**, a gente precisaria escrever algo do tipo:

`ExampleEntity resultado = exampleEntityRepository.findById(id);
    
    if (resultado != null) {
      return ExampleEntityDto.fromEntity(resultado);
    } else {
      throw new new EntityNotFoundException("Job not found for id: " + id);
    }`

...bem mais verboso e dependente de lógica manual, ou seja, mais suscetível a erros.